### PR TITLE
Bug 947305 - Parse SOF segements.

### PR DESCRIPTION
**Pull request by hfiguiere**

### CODE REVIEW

File: src/jpegParser.js [hunk 1]
```diff
@@ -74,15 +69,22 @@
       previousByte = currentByte;
       offset += 1;
     }
-    return offset + 1;
+    return offset - 1;
```
Comment (dmarcos) [inline src/jpegParser.js]: I need to refresh the details of the exif spec. Why did you replace - 1 with +1? I just want to make sure that it's not a mistake.

File: src/jpegSpec.js [hunk 2]
```diff
@@ -6,11 +6,10 @@
...
     0x02 : "RES",  // REServed ... (2-191) 0x02-0xbf
-    0xc0 : "SOF0", 0xc1 : "SOF1", 0xc2 : "SOF2",
-    0xc3 : "SOF3", 0xc5 : "SOF5", 0xc6 : "SOF6",
-    0xc7 : "SOF7", 0xc9 : "SOF8", 0xca : "SOF10",
-    0xcb : "SOF11", 0xcd : "SOF13", 0xce : "SOF14",
-    0xcf : "SOF15",
+    0xc0 : "SOF0", 0xc1 : "SOF1", 0xc2 : "SOF2", 0xc3 : "SOF3",
+    0xc5 : "SOF5", 0xc6 : "SOF6", 0xc7 : "SOF7",
+    0xc9 : "SOF9", 0xca : "SOF10", 0xcb : "SOF11",
+    0xcd : "SOF13", 0xce : "SOF14", 0xcf : "SOF15",
```
Comment (dmarcos) [inline src/jpegSpec.js]: Are these changes necessary?

File: src/jpegParser.js [hunk 3]
```diff
@@ -228,8 +241,13 @@
...
       var thumbnailMetaData = metaData && metaData.thumbnailMetaData;
       var thumbnailBlob = metaData && metaData.thumbnailBlob;
+      var imageSize = metaData && {
+        width: metaData.width,
+        height: metaData.height,
+        progressive: metaData.progressive
+      };
       metaData = metaData && metaData.Exif;
-      callback(error, metaData, thumbnailMetaData, thumbnailBlob);
+      callback(error, metaData, thumbnailMetaData, thumbnailBlob, imageSize);
```
Comment (dmarcos) [inline src/jpegParser.js]: Thi list of arguments is getting too long. Should we make an object with the data? callback(error, data) 
data {
exifMetadata : {}
 thumbnailMetaData : {}
thumbnailBlob: 
imageSize {}
}

File: src/jpegParser.js (hunk 1 above)
Comment (hfiguiere) [inline src/jpegParser.js]: Segment start at the 0xff marker. But we already read it on counted it. So we need to track back to include it.

This is not Exif, it is JPEG.

File: src/jpegParser.js (hunk 3 above)
Comment (hfiguiere) [inline src/jpegParser.js]: Sure. But I didn't want to break the API. If I do that I have to fix the documentation, the demo, the tests.

File: src/jpegSpec.js (hunk 2 above)
Comment (hfiguiere) [inline src/jpegSpec.js]: They bring clarity as I grouped the segment type in a consecutive manner. There is also a bug fix: 0xc9 is SOF9. There is no SOF8.
